src/app/3d/qgs3dmapcanvasdockwidget.cpp

```diff
@@ -223,6 +223,10 @@ void Qgs3DMapCanvasDockWidget::setMapSettings( Qgs3DMapSettings *map )
 
   connect( mCanvas->scene(), &Qgs3DMapScene::totalPendingJobsCountChanged, this, &Qgs3DMapCanvasDockWidget::onTotalPendingJobsCountChanged );
 
+  // Update the connection after the map is updated
+  mMapToolIdentify->updateSignalSlots();
+  mMapToolMeasureLine->updateSignalSlots();
```

Comment:
Would it make sense to emit a "mapSettingsChanged" signal from the mapCanvas and each tool can connect and react to this new signal internally?
This would also work if setMap is called directly on the canvas and will keep the API a bit tidier.

Reply:
hi @m-kuhn 
Thanks for the suggestion, I will update my PR for that.